internal/serverapi/serverapi.go

```diff
@@ -147,6 +147,31 @@ type CreateSnapshotSourceResponse struct {
 	SnapshotStarted bool `json:"snapshotted"` // whether snapshotting has been started
 }
 
+// DeletionResponseSingle contains response of attempted deletion of either snapshot or source.
+type DeletionResponseSingle struct {
+	Type    string `json:"type"`    // did we attempt to delete a snapshot or a source
+	Deleted bool   `json:"deleted"` // whether the object was deleted
+	Error   string `json:"error"`   // any errors
```

Comment:
errors are not normally returned inside the response but instead are serialized in ErrorResponse.

Reply:
Also, do we need this response at all? `Empty` may be sufficient.